MapboxNavigation/RouteTableViewController.swift

```diff
@@ -68,20 +68,41 @@ class RouteTableViewController: UIViewController {
 extension RouteTableViewController: UITableViewDelegate, UITableViewDataSource {
 
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+        return routeController.routeProgress.route.legs.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        // Don't display section header if there is only one step
+        guard routeController.routeProgress.route.legs.count > 1 else {
+            return nil
+        }
+        let leg = routeController.routeProgress.route.legs[section]
+        
+        let sourceName = leg.source.name
+        let destinationName = leg.destination.name
+        let legToFrom = leg.name.components(separatedBy: ", ")
+        
+        if let destinationName = destinationName?.nonEmptyString, legToFrom.count > 1 {
+            return "\(destinationName) via \(legToFrom.joined(separator: " and "))"
+        } else if let sourceName = sourceName?.nonEmptyString, let destinationName = destinationName?.nonEmptyString {
+            return "\(sourceName) anz \(destinationName)"
+        } else {
+            return leg.name
+        }
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return routeController.routeProgress.currentLeg.steps.count
+        return routeController.routeProgress.route.legs[section].steps.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: RouteTableViewCellIdentifier, for: indexPath) as! RouteTableViewCell
-        let leg = routeController.routeProgress.currentLeg
+        let legs = routeController.routeProgress.route.legs
+        
         
-        cell.step = leg.steps[indexPath.row]
+        cell.step = legs[indexPath.section].steps[indexPath.row]
         
-        if routeController.routeProgress.currentLegProgress.stepIndex + 1 > indexPath.row {
+        if indexPath.section < routeController.routeProgress.legIndex || (indexPath.section == routeController.routeProgress.legIndex && indexPath.row <= routeController.routeProgress.currentLegProgress.stepIndex) {
             cell.contentView.alpha = 0.4
```

Comment:
Upcoming steps may appear faded out if a passed step cell is being reused.
Consider adding an else statement or more succinctly `cell.contentView.alpha = didPassStep ? 0.4 : 1`